sdk/typescript/src/keypairs/zklogin/publickey.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fromB64 } from '@mysten/bcs';
+
+import { PublicKey } from '../../cryptography/publickey.js';
+import type { PublicKeyInitData } from '../../cryptography/publickey.js';
+import { SIGNATURE_SCHEME_TO_FLAG } from '../../cryptography/signature-scheme.js';
+import type { SerializedSignature } from '../../cryptography/signature.js';
+import { toPaddedBigEndianBytes } from '../../zklogin/utils.js';
+
+/**
+ * A zkLogin public identifier
+ */
+export class ZkLoginPublicIdentifier extends PublicKey {
+	private data: Uint8Array;
+
+	/**
+	 * Create a new ZkLoginPublicIdentifier object
+	 * @param value zkLogin public identifier as buffer or base-64 encoded string
+	 */
+	constructor(value: PublicKeyInitData) {
+		super();
+
+		if (typeof value === 'string') {
+			this.data = fromB64(value);
+		} else if (value instanceof Uint8Array) {
+			this.data = value;
+		} else {
+			this.data = Uint8Array.from(value);
+		}
+	}
+
+	/**
+	 * Checks if two zkLogin public identifiers are equal
+	 */
+	override equals(publicKey: ZkLoginPublicIdentifier): boolean {
+		return super.equals(publicKey);
```

Comment:
I think we probably want to have this throw an error instead.   I think the way this would currently be used is to check if 2 public key instances are for the same address

Reply:
revised